src/ripple/proto/ripple.proto

```diff
@@ -127,6 +127,23 @@ message TMCluster
     repeated TMLoadSource    loadSources     = 2;
 }
 
+// Request info on shards held
+message TMGetShardInfo
```

Comment:
high level question - why not implement this using strictly http methods? If each server provided its shard info (either in a new API or in server_info, for example) -- that combined with `peers` should be enough to gather this same data, right? 

Adding peer messages for something that is (1) strictly informational and (2) initiated by user request - seems less than ideal. If nodes needed to know the state of shards on its peers for operational/internal reasons, then it makes sense as part of the peer exchange, but seems like already had some of that info in the status message. Was that peer status message shard info not needed or not working out for some other reason? 

If we did provide the basic (per server) info via one or more RPC methods, then the crawler could be written in any language with a decent http/json client, and we could of course add a crawler method to our own RPC impl. if it's broadly useful.

Reply:
The old HTTPS crawl command provides shard info. Using that, a crawler can be written in any language and provide similar data.

Nodes need to know about the shards their peers hold. That information is used in the ledger acquire process for things like history backfilling. Knowing what the surrounding network has will also be beneficial when we enable IPFS, direct peer shard transfers or improving the index selection algorithm. We currently share the info via the status message, but it is not ideal. That design is limited and unable to discover shards beyond immediate peers. So the status field was migrated to a new message that overcomes the deficiency. But the actual shard index information is still packaged and used in the same manner. It is worth noting that peers automatically exchange shard info with each other upon connecting. They also update their peers with changes when necessary. Those exchanges are not initiated by user requests. The RPC may cause shard info caches to refresh but the info retrieved is utilized by the node. In actuality, this PR may be more about improving the method in which peers share their shard info on the network. The command allows sharing that info with users but at this time I can't say how useful that will be.

Reply:
To enable sharing shards via IPFS in a meaningful way, you'll need a stable representation of shard data first (see https://github.com/ripple/rippled/issues/2688), or you'll need to invest quite a bit of work into writing an IPFS native abstraction of a SHAmap (the relevant technique is called IPLD, there's examples for git or Ethereum available). If you go down that route, then this might actually obsolete the work on shards anyways, since all of history could be stored, accessed and shared that way.

I'd personally rather see a way to get rippled to sync a chain of ledger headers down to 32570 first (to be able to verify shard contents) before enabling them to be shared with peers or even peers of peers.

Reply:
@MarkusTeufelberger The `/crawl` endpoint has the data.